pkg/cri/resource-manager/policy/builtin/podpools/collector.go

```diff
@@ -0,0 +1,331 @@
+// Copyright 2020-2021 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podpools
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/intel/cri-resource-manager/pkg/cri/resource-manager/policy"
+	"github.com/intel/cri-resource-manager/pkg/metrics"
+	"github.com/intel/cri-resource-manager/pkg/sysfs"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics defines the metrics from policy level.
+type Metrics struct {
+	PoolMetrics map[string]*PoolMetrics
+}
+
+// PoolMetrics defines the metrics from pool level.
+type PoolMetrics struct {
+	DefName        string
+	PrettyName     string
+	CPUs           string
+	CPUMiliSize    string
+	Memory         string
+	ContainerNames string
+	PodNames       string
+}
+
+// Prometheus Metric descriptor indices and descriptor table
+const (
+	coreCpuUsageDesc = iota
+	poolCpuUsageDesc // count the pool CPU usage for podpools policy
+	numDescriptors   // descriptors total
+)
+
+var descriptors = []*prometheus.Desc{
+	coreCpuUsageDesc: prometheus.NewDesc(
+		"core_cpu_usage",
+		"CPU usage for a given core",
+		[]string{
+			"core_id",
+		}, nil,
+	),
+	poolCpuUsageDesc: prometheus.NewDesc(
+		"pool_cpu_usage",
+		"CPU usage for a given pool",
+		[]string{
+			"policy",
+			"pretty_name",
+			"def_name",
+			"CPUs",
+			"memory",
+			"pool_size",
+			"pod_name",
+			"container_name",
+		}, nil,
+	),
+}
+
+// core time statistic is used to calculate the CPU usage.
```

Comment:
I'd suggest following the godoc convention here, too: "...begins with the name of the element it describes".

```
// coreTimeStat stores stats and usage per CPU core.
```